mysqld_exporter.go

```diff
@@ -937,35 +937,38 @@ func scrapeSlaveStatus(db *sql.DB, ch chan<- prometheus.Metric) error {
 	}
 	defer slaveStatusRows.Close()
 
-	if slaveStatusRows.Next() {
-		// There is either no row in SHOW SLAVE STATUS (if this is not a
-		// slave server), or exactly one. In case of multi-source
-		// replication, things work very much differently. This code
-		// cannot deal with that case.
-		slaveCols, err := slaveStatusRows.Columns()
-		if err != nil {
-			return err
-		}
+	// Don't bother if there is no slave status row
+	if !slaveStatusRows.Next() {
+		return nil
+	}
 
-		// As the number of columns varies with mysqld versions,
-		// and sql.Scan requires []interface{}, we need to create a
-		// slice of pointers to the elements of slaveData.
-		scanArgs := make([]interface{}, len(slaveCols))
-		for i := range scanArgs {
-			scanArgs[i] = &sql.RawBytes{}
-		}
+	// There is either no row in SHOW SLAVE STATUS (if this is not a
+	// slave server), or exactly one. In case of multi-source
```

Comment:
If there is no row, then `slaveStatusRows.Next` will already fail I assume, so this comment needs to be moved up.